tools/astarte_export/lib/fetchdata/fetchdata.ex

```diff
@@ -0,0 +1,267 @@
+defmodule Astarte.Export.FetchData do
+  alias Astarte.Core.CQLUtils
+  alias Astarte.Core.Device
+  alias Astarte.Core.Mapping
+  alias Astarte.Core.Mapping.EndpointsAutomaton
+  alias Astarte.DataAccess.Database
+  alias Astarte.DataAccess.Interface
+  alias Astarte.DataAccess.Mappings
+  alias Astarte.Import
+  alias CQEx.Query
+  alias CQEx.Result
+  require Logger
+
+  defmodule State do
+    defstruct [
+      :device_id,
+      :revision,
+      :pending_empty_cache,
+      :secret_bcrypt_hash,
+      :first_registration,
+      :inhibit_request,
+      :cert_aki,
+      :cert_serial,
+      :first_credentials_request,
+      :last_credentials_request_ip,
+      :total_received_msgs,
+      :total_received_bytes,
+      :last_connection,
+      :last_disconnection,
+      :last_seen_ip,
+      :interfaces
+      ]
+   end 
+
+   
+   @spec get_connection(String.t()) :: {:ok, identifier()}
+ 
+   def get_connection(realm) do
+     {:ok, conn} = Database.connect(realm)
+     {:ok, conn}
+   end
+
+  
+  @spec get_devices(identifier()) :: list()  
+   
+  def get_devices(conn) do
+     statement = "SELECT * from devices;"
+     CQEx.Query.call!(conn, statement)
+     |> Enum.to_list
```

Comment:
I don't like this, we are going to keep in memory a potentially long list of devices in memory, instead of exporting device by device.